src/daos/MongoDAO.java

```diff
@@ -576,4 +577,35 @@ public void removeFriend(String userId, String friendId) {
         Document pullUserFromFriend = new Document("$pull", new Document("friends", new Document("username", userId)));
         userCollection.updateOne(friendQuery, pullUserFromFriend);
     }
+
+    @Override
+    public void addEventToGroup(Event event, String groupName) {
+        // Step 1: Query the group document using the groupName
+        Document query = new Document("groupname", groupName);
+        Document groupDoc = groupCollection.find(query).first();
+
+        // Step 2: Access the calendar document in the group
+        Document calendarDoc = (Document) groupDoc.get("calendar");
+
+
+        // Step 3: Get the existing list of events or initialize a new one
+        List<Document> eventDocs = (List<Document>) calendarDoc.get("events");
+        if (eventDocs == null) {
+            eventDocs = new ArrayList<>();
+        }
+
+        // Step 4: Create a new event document
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```

Comment:
This Formatter is called a lot in other methods. Please refractor it and make this a instance variable.